src/components/Calculator.js

```diff
@@ -1,41 +1,63 @@
-        <input type="text" className="input-screen" value="0" disabled />
+        <OutputResult outputvalue={outputValue} />
         <div className="magician-keys">
 
-          <button type="button" className="all-clear" value="all-clear">AC</button>
-          <button type="button" className="plus-minus-sign" value="+/-">+/-</button>
-          <button type="button" className="percentage-sign" value="%">%</button>
-          <button type="button" className="sign-operator" value="/">&divide;</button>
-
-          <button type="button" value="7">7</button>
-          <button type="button" value="8">8</button>
-          <button type="button" value="9">9</button>
-          <button type="button" className="sign-operator" value="*">&times;</button>
-
-          <button type="button" value="4">4</button>
-          <button type="button" value="5">5</button>
-          <button type="button" value="6">6</button>
-          <button type="button" className="sign-operator" value="-">-</button>
-
-          <button type="button" value="1">1</button>
-          <button type="button" value="2">2</button>
-          <button type="button" value="3">3</button>
-          <button type="button" className="sign-operator" value="+">+</button>
-
-          <button type="button" value="0" className="zero">0</button>
-          <button type="button" className="decimal" value=".">.</button>
-          <button type="button" className="sign-operator" value="=">=</button>
-
+          <MagicBtn value="AC" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="+/-" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="%" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="÷" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="7" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="8" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="9" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="x" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="4" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="5" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="6" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="-" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="1" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="2" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="3" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="+" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="0" clickEventListener={this.clickEventListener} />
+          <MagicBtn value="." clickEventListener={this.clickEventListener} />
+          <MagicBtn value="=" clickEventListener={this.clickEventListener} />
```

Comment:
**Changes Required**
- [ ] Good effort extracting the MagicBtn as it's own component. We can even do better, please let us try and store the button names into an array and use `Array.map` to display them dynamically. This will allow us to stick to the `DRY. principle` which will tidy our code.

Reply:
### I like that 👍 !